basic-integration/surfly-integration-with-tag-manager.md

```diff
@@ -0,0 +1,177 @@
+- Go to “versions” >> “actions” >> “review” to test the tag's functionality on your website
+- If everything is working properlyl, go to “versions” >> “actions” >> “publish” to make your tag go live
+
+{% em color="#ffffe0" %} For more info on Google Tag Manager, please visit the website https://www.google.com/analytics/tag-manager {% endem %}
+
+####The Tag
+
+Below, you'll find an example of how a tag, using the session-id approach, would look like. The example also includes the option of adding a third-party chat solution to your website. In this case we've used Olark.
+
+The tag is structured as follows:
+
+- First we add a div with an id of "footer". In the div we create two buttons: one to start a Surfly session and one to end it
+- Then we add the css by placing it between style tags
+- The last part of the tag is Javascript. First, we've made sure that GTM will show our buttons. Then we've added the Surfly snippet, the Surfly init function, a sessionStart() function, the Olark snippet and a sessionEnd() function.
+
+```javascript
+//Buttons to control the Surfly session
+<div class="footer" id="footer">
+  <button class="basic-button" id="btn-start-session" onclick="sessionStart()"><p id="id-cover">Get Help</p></button>
+  <button class="basic-button" id="btn-end-session" onclick="sessionEnd()">Stop sessie</button>
+</div>
+
+//Button styling
+<style type="text/css">
+  .basic-button {
+    background-color: #ffd05b;
+    width: 115px;
+    border: 1px solid #ffd05b;
+    border-radius: 33px;
+    height: 50px;
+    font-size: 14px;
+    color: #333333;
+    letter-spacing: 1px;
+    font-family: sans-serif;
+    font-weight: bold;
+    position: absolute;
+    left: 15px;
+    bottom: 15px;
+    box-shadow: 2px 2px rgba(0, 0, 0, 0.1);
+    position: fixed;
+  }
+
+  .basic-button:hover {
+    cursor: pointer;
+  }
+
+  .basic-button p {
+    margin-top: 14px;
+    font-size: 13px;
+  }
+
+  #btn-end-session {
+    display: none;
+  }
+</style>
+
+<script>
+//Make sure that GTM doesn't hide the buttons
+  document.getElementById('footer').parentElement.removeAttribute('style');
+  
+//Add the Surfly snippet
+  (function(s,u,r,f,l,y){s[f]=s[f]||{init:function(){s[f].q=arguments}};
+  l=u.createElement(r);y=u.getElementsByTagName(r)[0];l.async=1;
+  l.src='https://surfly.com/surfly.js';y.parentNode.insertBefore(l,y);})
+  (window,document,'script','Surfly');
+
+// Implement the Surfly init code, ADD YOUR OWN WIDGET KEY 
+  Surfly.init({widget_key:'** add your own Surfly widget key'}, function(init) {
+    if (init.success) {
+      if (Surfly.currentSession) {
+        // Laat binnen de sessie de btn-end-session zien
+        document.getElementById('btn-end-session').style.display="block";
+        // Zorg ervoor dat de btn-start-session ruimte maakt om de session-id te laten zien
+        document.getElementById("btn-start-session").style.visibility="hidden";
```

Comment:
I think it is better to make both buttons hidden by default, and then explicitly make them visible when needed. Like this:

```html
<style>
  #btn-end-session, #btn-start-session {
    display: none;
  }
</style>
```
and then in the code
```javascript
if (init.success) {
  if (Surfly.isInsideSession) {
    document.getElementById('btn-end-session').style.display = "block";
  } else {
    document.getElementById("btn-start-session").style.display = "block";
  }
}
```